meeting-minutes/2022-12-06.md

```diff
@@ -0,0 +1,319 @@
+
+cwebber: let's check who wants to do frequency in group chat of video
+
+cwebber: everyone has said monthly for now
+
+Jessica: monthly for now
+
+(Kris Kowal: +1 monthly or fortnightly)
+
+## Topic: Implementation summary
+
+cwebber: status in spritely:
+
+cwebber: very working racket implementation and getting close in guile
+
+cwebber: we have:
+- promise pipelining
+- asylic GC
+- handoffs (certificates)
+- and the usual things you have in CapTP otherwise.
+
+cwebber: we're hoping by next meeting we'll have guile and racket talking to each other
+
+Jessica: for agoric is it mark?  warner?
+
+warner: comms protocol from one kernel to another, we have a star configuration, one kernel can talk to another kernel through a comms protocol, it has promise pipelining and acyclic gc, there's a distinction between weakrefs and weak maps (?), not sure if that's a distinction for gc or not, don't have 3rd party handoff yet.  currently kernel assumes a safe way to connect kernels, we have vattp and captp division as usual
+
+danc: there are some smaller vattps
+
+markm: we do have a p2p protocol, and both vats talking to kernel, and one chain talking to other chains or non-chains.  we don't have chains talking to other chains yet
+
+markm: there are some hard problems to take on, one of the constraints of having an endpoint be a conventional blockchain is that a computationally conventional blockchain cannot keep secrets, so that has constrained the captp architecture in particular.  helps bootstrap from offline to online connectivity.  in certain conventions impossible to do a captp as a bearer right (sturdyrefs).  the ordering issues are important and probably
+
+markm: things we're going to need a variety, not a one-size-fits-all, and the thing we'll probably repeatedly revisit is distributed failure models... to what degree are failures masked or apparent for means for participants to recover from various failures
+
+Jessica: ian would you like to present on capn proto?
+
+isd: yes one difference is that we have substantial existing users and deployments so unfortunately we have a lot less flexibility in terms of change than others, but we currently have no implementation that does 3 party handoffs, there's some datatypes and specs in captp.capnp but that's likely to change in next few months.  have a client that wants various improvements in go handoffs
+
+isd: so that's interesting to me to make sure that we have better chance of interop
+
+isd: so I'd like to see if we have a way to shake out requirements early, so that's what's kind of going on on the go side
+
+isd: the other interesting thing within last week or so is that one of the forgefed people (forgefed attempting to do federated code forges) was planning on doing stuff with activitypub but has found that it's annoying to do anything but social pubsub, but they're interested, we wanted to possibly pull them in but they're in israel/palestine at the moment
```

Comment:
What got lost here is that Pere is actually prototyping stuff with haskell-capnp

Reply:
@zenhack maybe format your comment using the +/- diff thingy in such a way that @tsyesika can accept it with 1 button?